search-sentiments-map/test/trends-test.js

```diff
@@ -114,39 +93,40 @@ describe('Trends', function() {
       for (let i = 0; i < countryJson.length; i++) {
         mockTrendsByCountry.push({
           country: countryJson[i].id,
-          trends: [
-            {
-              topic: 'topic1',
-              traffic: '200K+',
-              exploreLink: 'exploreLink1',
-              articles: [
-                {
-                  title: 'article1',
-                  url: 'url1',
-                }, {
-                  title: 'article2',
-                  url: 'url2',
-                }
-              ],
-            }
-          ],
+          trends: 'mockTrends',
         });
       }
 
       // Stub the API call to googleTrends.
       sinon.stub(googleTrends, 'dailyTrends').resolves(mockDailyTrendsData);
 
-      // Stub the `saveTrendsAndDeletePrevious` function.
-      sinon.stub(trends, 'saveTrendsAndDeletePrevious')
-          .callsFake((trendsByCountry) => {
-        datastoreEntities.push({
-          key: 'fakeKey',
-          data: {
-            timestamp: Date.now(),
-            trendsByCountry: trendsByCountry,
-            globalTrends: trends.getGlobalTrends(trendsByCountry),
-          },
-        });
+      // Stub the `constructCountryTrendsJson` function because it is tested 
+      // separately.
+      sinon.stub(trends, 'constructCountryTrendsJson')
+          .callsFake((trendingSearches, countryCode) => {
+        return {
+          country: countryCode,
+          trends: 'mockTrends',
+        }
+      });
+
+      // Stub the `deleteAncientTrend` function because it is tested separately.
+      sinon.stub(trends, 'deleteAncientTrend')
+          .resolves('Not interested in the output');
+
+      // Stub the `getGlobalTrends` function because it is tested separately.
+      sinon.stub(trends, 'getGlobalTrends').callsFake(() => {
+        return 'mockGlobalTrends';
+      });
```

Comment:
Consider saving these stubs as variables and checking how many times they are called. Should be once each I think. This way you can make sure they are called the appropriate number of times.

Reply:
sure